Common/src/gRPC/Services/GrpcResultsService.cs

```diff
@@ -171,35 +173,41 @@ public override async Task<CreateResultsResponse> CreateResults(CreateResultsReq
-                                                                   ResultStatus.Created,
-                                                                   new List<string>(),
-                                                                   DateTime.UtcNow,
-                                                                   size,
-                                                                   Array.Empty<byte>());
+                                                 var (id, size) = await objectStorage_.AddOrUpdateAsync(new ObjectData
+                                                                                                        {
+                                                                                                          ResultId  = resultId,
+                                                                                                          SessionId = request.SessionId,
+                                                                                                        },
+                                                                                                        new List<ReadOnlyMemory<byte>>
+                                                                                                        {
+                                                                                                          rc.Data.Memory,
+                                                                                                        }.ToAsyncEnumerable(),
+                                                                                                        context.CancellationToken)
+                                                                                      .ConfigureAwait(false);
+
+                                                 return (new Result(request.SessionId,
+                                                                    resultId,
+                                                                    rc.Name,
+                                                                    "",
+                                                                    request.SessionId,
+                                                                    ResultStatus.Created,
+                                                                    new List<string>(),
+                                                                    DateTime.UtcNow,
+                                                                    size,
+                                                                    Array.Empty<byte>()), id);
                                                })
                                .WhenAll()
                                .ConfigureAwait(false);
 
-    await resultTable_.Create(results,
+    await resultTable_.Create(results.Select(tuple => tuple.Item1)
+                                     .AsICollection(),
```

Comment:
You can avoid materializing the list:
```suggestion
    await resultTable_.Create(results.ViewSelect(tuple => tuple.Item1),
```